qdlt/tests/test_qdltargument.cpp

```diff
@@ -0,0 +1,71 @@
+#include <gtest/gtest.h>
+
+#include <qdltargument.h>
+
+TEST(QDltArgument, constructor) {
+    QDltArgument arg;
+
+    ASSERT_EQ(arg.getTypeInfo(), QDltArgument::DltTypeInfoUnknown);
+    ASSERT_EQ(arg.getTypeInfoString(), QString{});
+    ASSERT_EQ(arg.getDataSize(), 0);
+    ASSERT_EQ(arg.toString(), QString("?")); // a bit weird for data size 0
+    ASSERT_EQ(arg.getValue(), QVariant());
+}
+
+TEST(QDltArgument, string_ascii) {
+    QDltArgument arg;
+    arg.setTypeInfo(QDltArgument::DltTypeInfoStrg);
+    const unsigned char asc_data[] = {
+        0xc3, 0xbc, 'b', 'e' // Ã¼be (and not übe)
+    };
+    arg.setData(QByteArray::fromRawData((const char*)asc_data, sizeof(asc_data)));
+    ASSERT_EQ(arg.toString(), QString::fromUtf8("Ã¼be"));
+
+    // parse from raw payload
+    const unsigned char asc_data2[] = {0x00, 0x02, 0x00, 0x00, // type info
+                                       0x04, 0x00,             // str len
+                                       0xc3, 0xbc, 'b',  'e'};
+    QByteArray payload = QByteArray::fromRawData((const char*)asc_data2, sizeof(asc_data2));
+    unsigned int offset = 0;
+    arg.setArgument(payload, offset, QDlt::DltEndiannessLittleEndian);
+    ASSERT_EQ(arg.getTypeInfo(), QDltArgument::DltTypeInfoStrg);
+    ASSERT_EQ(arg.getDataSize(), 4);
+    ASSERT_EQ(arg.toString().length(), 4);
+    ASSERT_EQ(arg.toString(), QString::fromUtf8("Ã¼be"));
```

Comment:
I think second implies the first

Reply:
yep, but I wanted to make it explicit. in the 2nd test the length for same binary payload is 3.